src/components/shared/modal/Modal.tsx

```diff
@@ -19,96 +19,62 @@ export interface IModalState {
   open: boolean;
 }
 
-@autobind
-class Modal extends React.Component<IModalProps, IModalState> {
-  public static defaultProps: IModalProps = {
-    closeButton: false,
-    open: false,
-  };
-
-  public state: IModalState = {
-    open: this.props.open,
-  };
-
-  public componentDidMount() {
-    this.bodyOverflow(this.state.open);
-  }
-
-  public componentWillUnmount() {
-    this.bodyOverflow(false);
-  }
-
-  public UNSAFE_componentWillReceiveProps(nextProps: IModalProps) {
-    if (nextProps.open !== this.props.open) {
-      this.setState({ open: nextProps.open });
-      this.bodyOverflow(nextProps.open);
-    }
-  }
-
-  public render() {
-    const { title, closeButton } = this.props;
-    const showHeader = !!title || closeButton;
-
-    if (!this.state.open) {
-      return null;
-    }
-
-    return (
-      <Portal>
-        <React.Fragment>
-          <Overlay open={this.state.open} onClick={this.close} />
-
-          <div
-            className={classNames("modal", this.props.className)}
-            role="dialog"
-            aria-modal="true"
-            aria-labelledby="modal-title"
-          >
-            {showHeader && (
-              <div className="modal__header">
-                {title && (
-                  <span id="modal-title" className="modal__header__title">
-                    {title}
-                  </span>
-                )}
-
-                {closeButton && (
-                  <IconButton
-                    className="modal__header__close-button"
-                    iconName="close"
-                    onClick={this.close}
-                  />
-                )}
-              </div>
-            )}
-
-            {this.props.children}
-          </div>
-        </React.Fragment>
-      </Portal>
-    );
-  }
-
-  private bodyOverflow(open: boolean) {
+const Modal: FC<IModalProps> = ({
+  open,
+  title,
+  closeButton,
+  onClose,
+  className,
+  ...props
+}) => {
```

Comment:
There is a [growing movement](https://github.com/typescript-cheatsheets/react#function-components) and [arguments](https://github.com/facebook/create-react-app/pull/8177) against using `React.FC`.  Prefer explicit function typings.

Reply:
> There is a [growing movement](https://github.com/typescript-cheatsheets/react#function-components) and [arguments](https://github.com/facebook/create-react-app/pull/8177) against using `React.FC`. Prefer explicit function typings.

I've been following the discussions as well and I do agree with the overal sentiment. Having that said. I've introduced it here explicitly for it's children type (because typing children is another touchy subject in typescript land). What would be your prefered way of typing components (in this case including a children type).

Reply:
Just include it in the props type:

```ts
type Props = {
  // other props...
  children?: React.ReactNode
}
```

Reply:
Is it bikeshedding or has pros/cons either way?

Reply:
I think mainly inference are the big pro's of the above way and adding children to things that don't need it. Updating the PR as we speak.